utils/build-presets.ini

```diff
@@ -534,6 +534,56 @@ swift-primary-variant-arch=x86_64
 # Don't build the benchmarks
 skip-build-benchmarks
 
+[preset: buildbot_incremental,tools=RA,stdlib=RD,smoketest=macosx,single-thread]
+mixin-preset=buildbot_incremental_base
+build-subdir=buildbot_incremental
+
+# We build release+asserts.
+release
+assertions
+
+# We run the OS X tests and validation tests.
+test
+validation-test
+
+llvm-targets-to-build=X86
+
+# Set the vendor to apple
+compiler-vendor=apple
+
+dash-dash
+
+# Always reconfigure cmake
+reconfigure
+
+# We want to always perform a verbose build
+verbose-build
+
+# Build ninja while we are at it
+build-ninja
+
+# We need to build the unittest extras so we can test
+build-swift-stdlib-unittest-extra
+
+# Make sure our stdlib is RA.
+swift-stdlib-build-type=RelWithDebInfo
+swift-stdlib-enable-assertions=true
+swift-stdlib-use-nonatomic-rc=true
+
+# Disable non-x86 building/testing.
+skip-build-ios
+skip-test-ios
+skip-build-tvos
+skip-test-tvos
+skip-build-watchos
+skip-test-watchos
+stdlib-deployment-targets=macosx-x86_64
+swift-primary-variant-sdk=OSX
+swift-primary-variant-arch=x86_64
+
+# Don't build the benchmarks
+skip-build-benchmarks
+
```

Comment:
Actually, since this preset for benchmarking purposes, we probably don't even need the debug info emission for the stdlib, as it only increases the build time.
So, one could even add:
`swift-stdlib-build-type=Release` after `dash-dash`